### PR TITLE
doc: create a new CQL reference section to aggregate CQL information

### PR DESCRIPTION
**Pull request by annastuchlik**
Fix https://github.com/scylladb/scylla-docs/issues/3733

This PR does not introduce any content changes in terms of CQL description or technical details.
The purpose of the update is to reorganize content to create a new high-level CQL section that can be used for CQL reference.

### CODE REVIEW

**Comment by tzach** [comment]
Many compile time warnings
```
docs/architecture/anti-entropy/read-repair.rst:71: WARNING: unknown document: /getting-started/ddl/
docs/architecture/anti-entropy/read-repair.rst:73: WARNING: unknown document: /getting-started/ddl/
docs/architecture/architecture-fault-tolerance.rst:22: WARNING: unknown document: /getting-started/cqlsh
docs/architecture/architecture-fault-tolerance.rst:77: WARNING: unknown document: /getting-started/consistency/
docs/architecture/compaction/compaction-strategies.rst:14: WARNING: unknown document: /getting-started/compaction
docs/architecture/compaction/compaction-strategies.rst:241: WARNING: unknown document: /getting-started/compaction
docs/architecture/raft.rst:91: WARNING: unknown document: ../getting-started/ddl
rst_include/apache-cql-return-index.rst:1: WARNING: unknown document: /getting-started/cql/
docs/cql/compaction.rst:327: WARNING: unknown document: /getting-started/cql/
docs/cql/cql-extensions.md:181: WARNING: None:any reference target not found: ./per-partition-rate-limit
docs/cql/cqlsh.rst:533: WARNING: unknown document: /getting-started/cql/
docs/cql/ddl.rst:88: WARNING: unknown document: /getting-started/reserved-keywords
docs/cql/ddl.rst:952: WARNING: unknown document: /getting-started/mv
docs/cql/ddl.rst:972: WARNING: unknown document: /getting-started/cql/
docs/cql/definitions.rst:178: WARNING: unknown document: /getting-started/ddl/
docs/cql/definitions.rst:179: WARNING: unknown document: /getting-started/dml/
rst_include/apache-cql-return-index.rst:1: WARNING: unknown document: /getting-started/cql/
docs/cql/dml.rst:146: WARNING: unknown document: /getting-started/time-to-live
docs/cql/dml.rst:769: WARNING: unknown document: /getting-started/time-to-live
docs/cql/dml.rst:955: WARNING: unknown document: /getting-started/cql/
rst_include/apache-cql-return-index.rst:1: WARNING: unknown document: /getting-started/cql/
docs/cql/json.rst:122: WARNING: unknown document: /getting-started/cql/
docs/cql/mv.rst:182: WARNING: unknown document: /getting-started/cql/
docs/cql/time-to-live.rst:105: WARNING: unknown document: /getting-started/cql/
rst_include/apache-cql-return-index.rst:1: WARNING: unknown document: /getting-started/cql/
docs/faq.rst:409: WARNING: unknown document: /getting-started/ddl/
docs/glossary.rst:31: WARNING: unknown document: /getting-started/consistency
docs/glossary.rst:34: WARNING: unknown document: /getting-started/consistency
docs/kb/compaction.rst:5: WARNING: unknown document: /getting-started/compaction
docs/kb/compaction.rst:53: WARNING: unknown document: /getting-started/compaction
docs/kb/compaction.rst:241: WARNING: unknown document: /getting-started/compaction
docs/kb/snapshots.rst:32: WARNING: unknown document: /getting-started/compaction
docs/kb/ttl-facts.rst:90: WARNING: unknown document: /getting-started/time-to-live/
<generated>:1: WARNING: unknown document: /getting-started/cqlsh
docs/operating-scylla/nodetool-commands/compact.rst:62: WARNING: unknown document: /getting-started/compaction
docs/operating-scylla/nodetool-commands/refresh.rst:8: WARNING: unknown document: /getting-started/mv/
docs/operating-scylla/nodetool-commands/refresh.rst:8: WARNING: unknown document: /getting-started/secondary-indexes/
docs/operating-scylla/security/authorization.rst:522: WARNING: unknown document: /getting-started/cql/
docs/operating-scylla/security/rbac-usecase.rst:215: WARNING: unknown document: /getting-started/cqlsh
docs/troubleshooting/copy-from-failed.rst:37: WARNING: unknown document: /getting-started/cql
operating-scylla/_common/tools_index.rst:2: WARNING: unknown document: /getting-started/cqlsh
docs/using-scylla/local-secondary-indexes.rst:117: WARNING: unknown document: /getting-started/secondary-indexes/
docs/using-scylla/secondary-indexes.rst:48: WARNING: unknown document: /getting-started/secondary-indexes
docs/using-scylla/secondary-indexes.rst:137: WARNING: unknown document: /getting-started/secondary-indexes/
docs/using-scylla/workload-prioritization.rst:67: WARNING: unknown document: /getting-started/reserved-keywords
```

File: docs/cql/cql-extensions.md [hunk 1]
```diff
@@ -178,5 +178,5 @@ If the driver doesn't support it, `Config_error` will be sent instead.
 
 For more details, see:
 
-- Detailed [`design notes`](./per-partition-rate-limit.md)
-- Description of the [rate limit exceeded](./protocol-extensions.md#rate-limit-error) error
+- Detailed [design notes](https://github.com/scylladb/scylla/blob/master/docs/dev/per-partition-rate-limit.md)
```
Comment (tzach) [inline docs/cql/cql-extensions.md:181]: suggestion (can be in a followup PR):
define a global variable for `https://github.com/scylladb/scylla/blob/master/docs/dev/` to simplify links to thde dev docs, and make the doc more robust for path updates.

See https://www.sphinx-doc.org/en/master/usage/extensions/extlinks.html which we already use

File: docs/_utils/redirects.yaml [hunk 2]
```diff
@@ -1,6 +1,23 @@
 ### a dictionary of redirections
 #old path: new path
 
+### moving the CQL reference files to the new cql folder
+
+/stable/getting-started/ddl.html: /stable/cql/ddl.html:
```
[the remark this answers is not in the document]
Comment (tzach) [reply docs/_utils/redirects.yaml:6]: also, should the redirects include the release (stable)?
This means `/5.0/getting-started/ddl`would not be redirected, which is wrong

File: docs/cql/cql-extensions.md (hunk 1 above)
Comment (annastuchlik) [reply docs/cql/cql-extensions.md:181]: @tzach I'll do it in this PR after David replies to your questions below.
I assume I may need to remove "stable", but leave ".html". The redirects without ".html" were not working (or I couldn't make them work).

File: docs/_utils/redirects.yaml [hunk 3]
```diff
@@ -1,6 +1,23 @@
...
+/getting-started/dml.html: /cql/dml.html
```
[the remark this answers is not in the document]
Comment (annastuchlik) [reply docs/_utils/redirects.yaml]: My bad. Fixed with the follow-up commit.

File: docs/_utils/redirects.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (annastuchlik) [reply docs/_utils/redirects.yaml]: Discussed with David - ".html" is required now due to the changes he made related to redirectons.

File: docs/cql/cql-extensions.md (hunk 1 above)
Comment (annastuchlik) [reply docs/cql/cql-extensions.md:181]: There are more links like that, so I think adding the variable should be a different PR. 
I've created an issue to do that: https://github.com/scylladb/scylla/issues/11073.

**Comment by nyh** [comment]
> @tzach I've addressed your comments. I've fixed one redirection, the rest should remain unchanged. @nyh This PR is ready to be merged, but there's a conflict. The types.rst page was updated with another PR and this PR moved that page to another location. Both changes are correct. What's the process for conflicts? What should I do next? I'm sorry for the trouble.

You can fix the conflict on your development machine by rebasing your branch on the newest master (`git fetch origin` followed by `git rebase origin/master`, and correct any conflicts that may arise like the change to the wrong file - it will give you instructions how), and then force-push the new branch back to the name place (`git push -f`).
You can also fix this more "manually" by dropping the problematic commit (`git rebase -i master` allows you to "drop" one or more commits) and redoing it manually by editing the file in its new location (or just overwriting it with a saved copy).

If you're worried you might break something, perhaps first make a copy of the files involved in case you lose something in the process :-)

**Comment by annastuchlik** [comment]
@nyh I've tried, but something's wrong. The docs look OK, but the build is pending. Should I close this PR and open a new one with the same changes?

**Comment by annastuchlik** [comment]
It looks like I messed up this PR. In addition, there are new conflicts now. I've opened a new PR with exactly the same updates, so I'm closing this PR. Please review and merge https://github.com/scylladb/scylla/pull/11083.

**Pull request closed without merging**